NostrGit/NostrGit
# Integrate cli tool by spearson78 for handling repos

### What does it do? Why?


Closes #115.

Integrates the [gitnostr](https://github.com/spearson78/gitnostr) repository to our codebase. Enables creating and cloning repositories with the git-nost-cli tool. The repositories are hosted on nostr relays. 

- Created a Dockerfile for the git-nostr-bridge.
- Created a docker-compose.yml to get git-nostr-bridge running conveniently.
- Added detailed instructions to readme.md

### QA


The following command (in project root)
```bash
$ docker compose up
```
should run a container called `git-nostr-bridge`. 

The user should now be able to use their git-nostr-cli with the following configuration:

```JSON
{
    "relays": ["<relay addresses>"],
    "privateKey": "<user's private key in hex>",
    "gitSshBase": "root@localhost"
}
```

### Review

- All GHA are success
- Use https://ui.shadcn.com/docs often as possible
- If Vercel build fails, run `yarn build` locally to get the same errors
      more to come...
